src/LineageTree/lineageTreeManager.py

```diff
@@ -0,0 +1,177 @@
+import os
+import pickle as pkl
+from functools import partial
+
+import numpy as np
+from edist.uted import uted
+
+from LineageTree import lineageTree
+
+
+class LineageTreeManager:
+    def __init__(self):
+        self.lineagetrees = {}
+        self.classification = {"Wt": {}, "Ptb": {}}
+        self.lineageTree_counter = 0
+        self.registered = {}
+
+    def get_next_tree(self):
+        self.lineageTree_counter += 1
+        return self.lineageTree_counter - 1
+
+    def add(
+        self, other_tree: lineageTree, name: str = "", classification: str = ""
+    ):
+        """Function that adds a new lineagetree object to the class.
+        Can be added either by .add or by using the + operator. If a name is
+        specified it will also add it as this specific name, otherwise it will
+        use the already existing name of the lineagetree. Lastly, it adds it to
+        the classification dict.
+
+        Args:
+            other_tree (lineageTree): Thelineagetree to be added.
+            name (str, optional): Then name of. Defaults to "".
+            classification (str, optional): If the user need to add it to classification
+                                    provide here the name of the class.
+                                    Defaults to "".
+
+        Returns:
+            _type_: _description_
+        """
+        for tree in self.lineagetrees.values():
+            if tree == other_tree:
+                return False
+        if isinstance(other_tree, lineageTree):
+            if name:
+                self.lineagetrees[name] = other_tree
+            else:
+                try:
```

Comment:
Don't do a try except when you don't know how to do it.
Here is a solution:
```python
if hasattr(other_tree, "name"):
    self.lineagetrees[other_tree.name] = other_tree
else:
    ...
```

Reply:
Here I meant "don't do try except just because you don't know how to do it otherwise"
😁